pkg/builder/controller.go

```diff
@@ -66,15 +69,16 @@ type Builder struct {
 
 // ControllerManagedBy returns a new controller builder that will be started by the provided Manager.
 func ControllerManagedBy(m manager.Manager) *Builder {
-	return &Builder{mgr: m}
+	return &Builder{cluster: m, mgr: m}
 }
 
 // ForInput represents the information set by the For method.
 type ForInput struct {
-	object           client.Object
-	predicates       []predicate.Predicate
-	objectProjection objectProjection
-	err              error
+	object              client.Object
+	forceDefaultCluster bool // in cluster-aware mode, force the object to be watched in the default cluster
```

Comment:
So its set true in cluster-aware mode or other way around?

Reply:
actually I reverted that, but left some bread crumps here. Not sure we really want it. To be discussed.